nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/logback.xml

```diff
@@ -30,10 +30,13 @@
               To GZIP rolled files, replace '.log' with '.log.gz'.
               To ZIP rolled files, replace '.log' with '.log.zip'.
             -->
-            <fileNamePattern>${org.apache.nifi.bootstrap.config.log.dir}/nifi-app_%d{yyyy-MM-dd_HH}.%i.log</fileNamePattern>
+            <fileNamePattern>${org.apache.nifi.bootstrap.config.log.dir}/nifi-app_%d{yyyy-MM-dd_HH}.%i.log.gz</fileNamePattern>
             <maxFileSize>100MB</maxFileSize>
             <!-- keep 30 log files worth of history -->
             <maxHistory>30</maxHistory>
+            <totalSizeCap>100GB</totalSizeCap>
```

Comment:
if the default is 100MB per file and keeping 30 files then 3GB seems like a reasonable default cap size.  Am I reading that right?

Reply:
`3GB` is fine here. I'm not married to any of these values, I was just erring on the side of caution.

Reply:
Actually "keep 30 log files worth of history" is a bit misleading.  maxHistory ignores the %i part of the fileNamePattern.  If your pattern is "nifi-app_%d{**yyyy-MM-dd_HH**}.%i.log" then it's maxHistory hours, and a pattern of "nifi-app_%d{**yyyy-MM-dd**}.%i.log" is maxHistory days.  You can rollover with the %i pattern hundreds of times and maxHistory doesn't care.
Regardless, I'm in favor of a smaller default totalSizeCap instead of 100GB.  nifi-registry and stateless appear to use 10GB here.  I have no idea if totalSizeCap respects the %i pattern.